### PR TITLE
Robot Quiz

### PR DESCRIPTION
**Pull request by daynewright**

### CODE REVIEW

**Comment by daynewright** [comment]
@gregkorte I am ready for review.  I would like it to be cleaner but it works. Also, I am not sure if my app structure is anything close to best practice but it initially made sense to me.

I used `npm install` and `bower install` to get the dependencies.  Then I have`gulp watch-reload` for live reload and babel.  Finally `gulp build` makes a (hopefully) deployable version.

Here is the structure I used:

<pre>
<b>src</b>
 |- <b>build</b> <i>(created from <b>gulp build</b> for deployment with concat/minify for js/css/html)</i>
 |- <b>dev</b> <i>(all created files)</i>
    |- <b>babel</b> <i>(created from <b>gulp babel-js</b>.  index.html in dev links to these js files)</i>
    |- <b>css</b>
    |-<b>images</b> 
    |- <b>js</b>
 |- <b>lib</b>
 |- <b>spec</b> <i>(jasmine test js files) </i>
 |- <b>test.html</b> <i>(jasmine test html file calling files in /spec)</i>
</pre>

Here are the gulp tasks I have:

<pre>
  <b>gulp babel-js</b> <i>(creates js files needed for index if not already created)</i>
  <b>gulp lint</b> <i>(runs jslint on all dev/js files)</i>
  <b>gulp watch-reload</b> <i>(runs gulp watch with babel build and live reload on port 8081)</i>
  <b>gulp build</b> <i>(creates src/build folder with minify, concat and inject tags for new html.) </i>
</pre>

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+// Lint task for JS source files
+
+// Babel task for JS source files
+ gulp.task('babel-js',() =>
+    gulp.src(files.js.scripts)
+      .pipe(babel({presets: ['es2015']}))
+      .pipe(gulp.dest('src/dev/babel'))
+ );
+ // Babel task for JS source files
+
```
Comment (JoeShep) [inline gulpfile.js:126]: Any reason you're using Babel? At first glace I don't see any ES6 code that wouldn't work in the Browser

File: src/dev/babel/attack.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    if (arena.getRobotPlayer().health <= 0) {
+      youLose();
+    }
+    if (arena.getRobotComp().health <= 0) {
+      youWin();
+    }
+  };
+
+  function youWin() {
+    swal({
```
Comment (JoeShep) [inline src/dev/babel/attack.js:37]: Had not heard of sweet alert. Very cool.

File: src/dev/babel/weapons.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var Arena = function (arena) {
+  arena.Arsonal = {};
+
```
Comment (JoeShep) [inline src/dev/babel/weapons.js:5]: (sp) Arsenal. Couldn't resist. Former editor

File: src/dev/babel/attack.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    var enemyId = this === arena.getRobotPlayer() ? 1 : 0;
+
+    arena.buildDomElements();
+
+    var $health = $('#health-' + enemyId);
+    var $image = $('#image-' + enemyId);
+
+    $($health[0]).addClass('flash');
+    $($image[0]).addClass('shake');
```
Comment (JoeShep) [inline src/dev/babel/attack.js:17]: Nice touch

**Comment by JoeShep** [comment]
Hey Dayne, this project runs really well. The UI has some nice touches, and I like the use of that alert library you implemented. You've use the revealing pattern very well, you have a great grasp of setting up iifes to create namespaced modules. Good job. You're also showing comfort with prototypal inheritance, which is the real key here.

As you might have guessed, my only ding is for your file structure. It's not anything I've seen before and it took me a bit to even find the index file so I could run the web server. Is the lib folder where it is to serve both the dev and the build directories? If not, I would put it inside the dev folder. It's also standard to have your index file at the root. Interesting experiment, but I would stick to something less obtuse, especially if you're working on a project with a team. 

Overall, though, this is really strong and I think you're in great shape moving forward. Looking forward to seeing more of your work.

File: gulpfile.js (hunk 1 above)
Comment (daynewright) [reply gulpfile.js:126]: At the time I thought if you have ANY ES6 then you need Babel.

File: src/dev/babel/weapons.js (hunk 3 above)
Comment (daynewright) [reply src/dev/babel/weapons.js:5]: If that is the only misspelled word then I consider it a WIN! 😃

**Comment by daynewright** [comment]
The app structure was a complete guess at the time and looking back I honestly can't tell you why I did what I did. Ha!  I was attempting to use gulp to minify and build everything and did not have a clear idea of how to do it so I just went with what I could think of at the time.   I wanted to get more experience with using gulp and what you can do with it.  

After the last few weeks (and being introduced to Angular) I think I have a better idea of how to structure an app from this point moving forward.  

Thanks for the detailed review!

**Pull request closed without merging**